### PR TITLE
[maddy] Initial release

### LINKED ISSUE #1384: [maddy] Add new email chart for simple but full email server
# Details

**Helm chart name:**

maddy

**Describe the solution you'd like:**

I am working in an email server chart that is k8s-at-home friendly:
- simple to configure via helm
- can run in cluster mode
- provide IMAP
- support required security standards: DMARC, DKIM, SPF, STARTTL, external relays (I might need to send with AWS SES or similar if my home IP is marked as spam)

After considering Mailu, main-in-a-box, Mailcow and Wildduck I decided to go with Maddy as I like the single-binary instead of having to deploy multiple services and connect them together. I was really tempted by Wildduck due to the dedicated webmailer but having to configure the incoming/outgoing SMTP server and the missing documentation for running stand alone moved the neddle for Maddy.

I also plan to combine with the calendar in nextcloud so I will use its webclient. For tablets and cellphones I use the native IMAP clients.

**Anything else you would like to add:**

I have a working PoC chart working - I will clean up and provide a PR soon.

**Additional Information:**

Related to https://github.com/k8s-at-home/charts/issues/1218

### PR DESCRIPTION
**Pull request by angelnu**
**Description of the change**

Add [Maddy](https://maddy.email) a lightweight email server.

**Benefits**

Very light email server (single binary, less than 6 MB of RAM) but still including all the required security features (DMARC, DKIM, SPF, STARTLS, etc) and protocols (IMAP and SMTP).

Quoting from the project

> Maddy Mail Server implements all functionality required to run a e-mail server. It can send messages via SMTP (works as MTA), accept messages via SMTP (works as MX) and store messages while providing access to them via IMAP. In addition to that it implements auxiliary protocols that are mandatory to keep email reasonably secure (DKIM, SPF, DMARC, DANE, MTA-STS).
> 
> It replaces Postfix, Dovecot, OpenDKIM, OpenSPF, OpenDMARC and more with one daemon with uniform configuration and minimal maintenance cost.
> 
> Note: IMAP storage is "beta". If you are looking for stable and feature-packed implementation you may want to use Dovecot instead. maddy still can handle message delivery business.

**Possible drawbacks**

I was not able to run `./hack/gen-helm-docs.sh stable maddy`:

```
-] Copying templates to /root/charts/charts/stable/maddy
INFO[2022-01-23T22:28:13Z] Found Chart directories [.]                  
INFO[2022-01-23T22:28:13Z] Generating README Documentation for chart /root/charts/charts/stable/maddy 
WARN[2022-01-23T22:28:13Z] Error generating gotemplates for chart /root/charts/charts/stable/maddy: template: /root/charts/charts/stable/maddy:120: function "fromYaml" not defined 
```

and the README.md was not updated.

**Applicable issues**

- fixes #1384 

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Title of the PR starts with chart name (e.g. `[home-assistant]`)
- [x] Variables are documented in the README.md (this can be done with using our helm-docs wrapper `./hack/gen-helm-docs.sh stable <chart>`)

### CODE REVIEW

**Comment by angelnu** [comment]
@bjw-s Do you know what the issue might be? I updated to the latest helm-docs and I use golang 1.17 in my Linux dev box.

I uploaded the PR sources to a dev branch in case you could re-generate the README.md for the maddy chart.

**Comment by angelnu** [comment]
The pipeline updated the `README.md` but I would still want to be able to get it working again on local to fix problems before the PR....

**Comment by bjw-s** [comment]
We have recently changed the dependency for helm-docs to use our own branch (we are still thinking of how we can neatly integrate our change in to the upstream repo):

https://docs.k8s-at-home.com/our-helm-charts/development/creating-a-new-chart/#dependencies

**Comment by angelnu** [comment]
Thanks @bjw-s , I had forgotten about our fork x-) I updated and it works now!

File: charts/stable/maddy/templates/common.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+{{/* Append the hardcoded settings */}}
+{{- define "maddy.harcodedValues" -}}
+
+configmap:
+  config:
+    enabled: true
+    data:
+      maddy.conf: |
+        {{ tpl (.Files.Get "files/maddy.conf") . | nindent 8 }}
```
Comment (bjw-s) [inline charts/stable/maddy/templates/common.yaml:34]: This is very nifty 👍 💯

File: charts/stable/maddy/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+apiVersion: v2
+appVersion: 1.0.0
```
Comment (bjw-s) [inline charts/stable/maddy/Chart.yaml:2]:
```suggestion
appVersion: v0.5.2
```

File: charts/stable/maddy/values.yaml [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+#
+# This chart inherits from our common library chart. You can check the default values/options here:
+# https://github.com/k8s-at-home/library-charts/tree/main/charts/stable/common/values.yaml
+#
+
+image:
+  # -- image repository
+  repository: foxcpp/maddy
+  # -- image tag
+  tag: v0.5.2
```
Comment (bjw-s) [inline charts/stable/maddy/values.yaml:12]:
```suggestion
  # -- image tag
  # @default -- chart.appVersion
  tag:
```

In new charts we are trying to define the appVersion only once, that way we "force" ourselves to update the appVersion in Chart.yaml as well 😅

**Review by bjw-s — CHANGES_REQUESTED** [review]
Thanks for yet another PR! Looks good! Only a small change requested with regard to the `appVersion`.

File: charts/stable/maddy/templates/common.yaml (hunk 1 above)
Comment (angelnu) [reply charts/stable/maddy/templates/common.yaml:34]: :-)

I also do it with the rules set - much easier to read than embedding in templates :-)

**Comment by angelnu** [comment]
Accepted your changes (missed changing the appversion in the template - was too late yesterday night :-D ). So ready to merge once CI is done.

Now we can start tuning the rules - I just included the default setup but I expect to play more with it. Hopefully others can also come with their own configs to share.

**Review by bjw-s — APPROVED** [review]

**Pull request merged**